src/graphql.tsx

```diff
@@ -573,8 +573,16 @@ export default function graphql(
             // while loading, we should use any previous data we have
             assign(data, this.previousData, currentResult.data);
           } else {
-            assign(data, currentResult.data);
-            this.previousData = currentResult.data;
+            let dataToMerge = currentResult.data;
+            // if we have an error, include any previously cached data
+            if (error) {
+              let previousResult = this.queryObservable.getLastResult();
+              if (previousResult && !previousResult.error) {
+                dataToMerge = assign({}, previousResult.data, currentResult.data);
```

Comment:
Hey @helfer thanks for the review.  I'm happy you're encouraged by this PR.  Please give me a few days to update the PR as I'm swamped on a few higher priority issues.  I'll get the appropriate testes in there and clean up the currentResult stuff and just use getLastResult().

Thanks again!